File: 70_ClimbingStairs.py
class Solution:
    cache = {0:1, 1:1}
    def climbStairs(self, n: int) -> int:
        if n in self.cache:
            return self.cache[n]
        else:
            self.cache[n] = self.climbStairs(n-1) + self.climbStairs(n-2)
            return self.cache[n]

File: java/70_ClimbingStairs.java
class Solution {
    HashMap<Integer, Integer> memo = new HashMap<>();
    public int climbStairs(int n) {
        if (n <= 1) return 1;
        else if (memo.get(n) != null) return memo.get(n);
        
        memo.put(n, climbStairs(n-1) + climbStairs(n-2));
        return memo.get(n);
    }
}
